ee/cis/macos-13/cis-policy-queries.yml

```diff
@@ -1306,6 +1306,90 @@ spec:
+  query: |
+    SELECT 1 WHERE NOT EXISTS (
+    SELECT 1 FROM file WHERE 
+      path LIKE '/Applications/%%'  
+      AND type = 'directory' 
+      AND directory LIKE '%.app' 
+      AND   SUBSTRING( mode ,-1) & 0x2 !=0 -- mode's last char is others' permissions. bitwise with 0x2 means write permissions. (which we do not want here) 
+    );
+  purpose: Informational
+  tags: compliance, CIS, CIS_Level1, CIS5.1.5
+  contributors: sharon-fdm
+---
+apiVersion: v1
+kind: policy
+spec:
+  name: CIS - Ensure No World Writable Files Exist in the System Folder
+  platforms: macOS
+  platform: darwin
+  description: |
+    Software sometimes insists on being installed in the /System/Volumes/Data/System Directory and has inappropriate world-writable permissions.
+    Macs with writable files in System should be investigated forensically. A file with open writable permissions is a sign of at best a rogue application. It could also be a sign of a computer compromise and a persistent presence on the system.
+  resolution: |
+    Ask your system administrator to deploy a script that will ensure folders are not world-writable in the /System folder. 
+    /usr/bin/sudo IFS=$'\n'
+    for sysPermissions in $( /usr/bin/find /System/Volumes/Data/System -type d -perm -2 | /usr/bin/grep -v "Drop Box" ); 
+    do
+      /bin/chmod -R o-w "$sysPermissions"
+    done
+  query: |
+    SELECT 1 WHERE NOT EXISTS (
+    SELECT 1 FROM file WHERE 
+      path LIKE '/System/Volumes/Data/System/%%'  
+      AND type = 'directory' 
+      AND directory NOT LIKE '%Drop Box%' 
+      AND   SUBSTRING( mode ,-1) & 0x2 !=0 -- mode's last char is others' permissions. bitwise with 0x2 means write permissions. (which we do not want here) 
+    );
+  purpose: Informational
+  tags: compliance, CIS, CIS_Level1, CIS5.1.6
+  contributors: sharon-fdm
+---
+apiVersion: v1
+kind: policy
+spec:
+  name: CIS - Ensure No World Writable Files Exist in the Library Folder
+  platforms: macOS
+  platform: darwin
+  description: |
+    Software sometimes insists on being installed in the /System/Volumes/Data/Library Directory and has inappropriate world-writable permissions.  
+    Folders in /System/Volumes/Data/Library should not be world-writable. The audit check excludes the /System/Volumes/Data/Library/Caches and /System/Volumes/Data/Library/Preferences/Audio/Data folders where the sticky bit is set.
+  resolution: |
+    Ask your system administrator to deploy a script that will ensure folders are not world-writable in the /System folder. 
+    /usr/bin/sudo IFS=$'\n'
+    for libPermissions in $( /usr/bin/find /System/Volumes/Data/Library -type d -perm -2 | /usr/bin/grep -v Caches | /usr/bin/grep -v /Preferences/Audio/Data); 
+    do
+      /bin/chmod -R o-w "$libPermissions"
+    done
+  query: |
+    SELECT 1 WHERE NOT EXISTS (
+    SELECT 1 FROM file WHERE 
+      path LIKE '/System/Volumes/Data/System/%%'  
+      AND type = 'directory' 
+      AND directory NOT LIKE '%Caches%' 
```

Comment:
I believe this is too broad, what we want is to exclude folders `/System/Volumes/Data/Library/Caches` and `/System/Volumes/Data/Library/Preferences/Audio/Data/` only, so maybe:
```sql
... AND path NOT LIKE '/System/Volumes/Data/Library/Caches/%%'
AND path NOT LIKE '/System/Volumes/Data/Library/Preferences/Audio/Data/%%'
```

Reply:
In the CIS doc they exclude ALL appearances.


`
/usr/bin/sudo /usr/bin/find /System/Volumes/Data/Library -type d -perm -2 -
ls | /usr/bin/grep -v Caches | /usr/bin/grep -v /Preferences/Audio/Data |
/usr/bin/wc -l | /usr/bin/xargs
`
Note:
`grep -v Caches`
`grep -v /Preferences/Audio/Data`
 Tell me what you think.

Reply:
Ah, I see what you mean.

As you say, the terminal method has `grep -v Caches` and `grep -v /Preferences/Audio/Data`, but the rationale says:
```
The audit check excludes the /System/Volumes/Data/Library/Caches and /System/Volumes/Data/Library/Preferences/Audio/Data folders where the sticky bit is set.
```

Maybe all `Caches` directories should be skipped (e.g.  I can see`/System/Volumes/Data/Library/User\ Template/Non_localized/Library/Caches` on my system). Let's trust the terminal method then (which means the query is a-ok).